Doc update (#667)
oneapi-src/oneDAL: docs/source/dev_guide/algorithms/svm/support-vector-machine-classifier.rst

Training Stage
--------------

|short_name| provides two methods to train the SVM model:

> - Boser method [Boser92]_

Reviewer: Need to add alternative names for clarification

Author: what are the alternative names?

Reviewer: I mean SMO (Sequential Minimal Optimization), but some addition clarification about difference between Boser's and Platt's algorithms. I suggest to consult with @PetrovKP about better formulation. He is ready to help with this.

Reviewer 2: I would suggest describing:

```
Boser method [Boser92] which is being decided the Sequential Minimal Optimization algorithm proposed Platt
```

Author: added description

> - Boser method [Boser92]_ that is the Sequential Minimal Optimization algorithm proposed by Platt [Platt98]_

Reviewer: @PetrovKP , add details about differences between Boser and Platt. According to your formulation, they are the same, but it isn't true. Boser in '92 proposed something different then Platt in '98.

Reviewer 2: ```suggestion
- Boser method [Boser92]_ - performance-oriented variant of Boser [Boser92] and Platt [Platt98] algorithms.
```

At the training stage, SVM classifier has the following parameters:

.. list-table::
   :header-rows: 1
   :widths: 10 10 60
   :align: left

   * - Parameter
     - Default Value
     - Description
   * - ``algorithmFPType``
     - ``float``
     - The floating-point type that the algorithm uses for intermediate computations. Can be float or double.
   * - ``method``
     - ``defaultDense``
     - The computation method used by the SVM classifier. Available methods for the training stage:

       For CPU:

>         - ``defaultDense`` – Boser method [Boser92]_

Author: I'll add this closer to beta08, since right now it's not supported yet

>         - ``defaultDense`` – Boser method [Boser92]_

Reviewer 2: Ok

       For GPU:

        - ``thunder`` - Thunder method [Wen2018]_

   * - ``nClasses``
     - :math:`2`
     - The number of classes.
   * - ``C``
     - :math:`1`
     - The upper bound in conditions of the quadratic optimization problem.
   * - ``accuracyThreshold``
     - :math:`0.001`
     - The training accuracy.
   * - ``tau``
     - :math:`1.0e-6`
     - Tau parameter of the WSS scheme.
   * - ``maxIterations``
     - :math:`1000000`
     - Maximal number of iterations for the algorithm.
   * - ``cacheSize``
     - :math:`8000000`
     - The size of cache in bytes for storing values of the kernel matrix.
       A non-zero value enables use of a cache optimization technique.
   * - ``doShrinking``
     - ``true``
     - A flag that enables use of a shrinking optimization technique.

>        .. note:: This parameter is only supported for ``defaultDense`` method.

Author: There is more variety there. I'd argue that one small difference between two method doesn't justify creation of a new column. If you disagree, I'll add it, of course.

>        .. note:: This parameter is only supported for ``defaultDense`` method.

Reviewer 2: I think it should be consistent. And in the future, specific parameters for each method may be added. But you can leave it now. :)

At the prediction stage, SVM classifier has the following parameters:

.. list-table::
   :header-rows: 1
   :widths: 10 10 60
   :align: left

   * - Parameter
     - Default Value
     - Description
   * - ``algorithmFPType``
     - ``float``
     - The floating-point type that the algorithm uses for intermediate computations. Can be float or double.
   * - ``method``
     - ``defaultDense``
     - Performance-oriented computation method, the only prediction method supported by the algorithm.
   * - ``nClasses``
     - :math:`2`
     - The number of classes.
   * - ``kernel``
     - Pointer to object of the ``KernelIface`` class

>      - The kernel function. By default, the algorithm uses a linear kernel.

Reviewer 2: Please, update examples:

```
  .. tab:: DPC++
   Batch Processing:
   -  :ref:`svm_two_class_dense_batch.cpp>`

  .. tab:: C++
   Batch Processing:
   -  :cpp_example:`svm_two_class_boser_dense_batch.cpp <svm/svm_two_class_boser_dense_batch.cpp>`
   -  :cpp_example:`svm_two_class_boser_csr_batch.cpp <svm/svm_two_class_boser_csr_batch.cpp>`
   -  :cpp_example:`svm_two_class_thunder_dense_batch.cpp <svm/svm_two_class_thunder_dense_batch.cpp>`
   -  :cpp_example:`svm_two_class_thunder_csr_batch.cpp <svm/svm_two_class_thunder_csr_batch.cpp>`

.. tab:: Java*
   Batch Processing:
   -  :java_example:`SVMTwoClassBoserDenseBatch.java <svm/SVMTwoClasBosersDenseBatch.java>`
   -  :java_example:`SVMTwoClassBoserCSRBatch.java <svm/SVMTwoClassBoserCSRBatch.java>`
   -  :java_example:`SVMTwoClassThunderDenseBatch.java <svm/SVMTwoClassThunderDenseBatch.java>`
   -  :java_example:`SVMTwoClassThunderCSRBatch.java <svm/SVMTwoClassThunderCSRBatch.java>`
```

Author: fixed, thanks

At the prediction stage, SVM classifier has the following parameters:

.. list-table::
   :header-rows: 1
   :widths: 10 10 60
   :align: left

   * - Parameter
     - Default Value
     - Description
   * - ``algorithmFPType``
     - ``float``
     - The floating-point type that the algorithm uses for intermediate computations. Can be float or double.
   * - ``method``
     - ``defaultDense``
     - Performance-oriented computation method, the only prediction method supported by the algorithm.
   * - ``nClasses``
     - :math:`2`
     - The number of classes.
   * - ``kernel``
     - Pointer to object of the ``KernelIface`` class
     - The kernel function. By default, the algorithm uses a linear kernel.

Examples
--------

.. tabs::

  .. tab:: DPC++

    Batch Processing:
    

>     -  :ref:`svm_two_class_dense_batch.cpp`

Reviewer 2: I renamed an example today https://github.com/oneapi-src/oneDAL/pull/696
```suggestion
    -  :ref:`svm_two_class_thunder_dense_batch.cpp`
```

Author: fixed
